File: QA-project/backend/app/models/otp_codes.py
from sqlalchemy import Column, Integer, String, DateTime, func
from app.database import Base


class OTPCode(Base):
    __tablename__ = "otp_codes"

    otp_id = Column(Integer, primary_key=True, autoincrement=True)
    email = Column(String(255), nullable=False, index=True)
    otp_code = Column(String(6), nullable=False)
    purpose = Column(String(50), nullable=False)  # "registration" or "reset_password"
    expires_at = Column(DateTime, nullable=False)
    is_used = Column(Integer, nullable=False, server_default="0")  # 0=unused, 1=used
    created_at = Column(DateTime, server_default=func.now(), nullable=False)

File: QA-project/backend/src/db/schema/otp-codes.ts
import {
  pgTable,
  serial,
  varchar,
  integer,
  timestamp,
} from "drizzle-orm/pg-core";

export const otpCodes = pgTable("otp_codes", {
  otpId: serial("otp_id").primaryKey(),
  email: varchar("email", { length: 255 }).notNull(),
  otpCode: varchar("otp_code", { length: 6 }).notNull(),
  purpose: varchar("purpose", { length: 50 }).notNull(),
  expiresAt: timestamp("expires_at").notNull(),
  isUsed: integer("is_used").notNull().default(0),
  createdAt: timestamp("created_at").defaultNow().notNull(),
});
